controllers/operatorpolicy_controller.go

```diff
@@ -358,98 +359,106 @@ func buildOperatorGroup(
 	return operatorGroup, nil
 }
 
-func (r *OperatorPolicyReconciler) handleSubscription(ctx context.Context, policy *policyv1beta1.OperatorPolicy) error {
+func (r *OperatorPolicyReconciler) handleSubscription(
+	ctx context.Context, policy *policyv1beta1.OperatorPolicy,
+) (*operatorv1alpha1.Subscription, error) {
 	watcher := opPolIdentifier(policy.Namespace, policy.Name)
 
 	desiredSub, err := buildSubscription(policy)
 	if err != nil {
-		return fmt.Errorf("error building subscription: %w", err)
+		return desiredSub, fmt.Errorf("error building subscription: %w", err)
 	}
 
 	foundSub, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, desiredSub.Namespace, desiredSub.Name)
 	if err != nil {
-		return fmt.Errorf("error getting the Subscription: %w", err)
+		return desiredSub, fmt.Errorf("error getting the Subscription: %w", err)
 	}
 
 	if foundSub == nil {
 		// Missing Subscription: report NonCompliance
 		err := r.updateStatus(ctx, policy, missingWantedCond("Subscription"), missingWantedObj(desiredSub))
 		if err != nil {
-			return fmt.Errorf("error updating status for a missing Subscription: %w", err)
+			return desiredSub, fmt.Errorf("error updating status for a missing Subscription: %w", err)
 		}
 
 		if policy.Spec.RemediationAction.IsEnforce() {
 			err := r.Create(ctx, desiredSub)
 			if err != nil {
-				return fmt.Errorf("error creating the Subscription: %w", err)
+				return desiredSub, fmt.Errorf("error creating the Subscription: %w", err)
 			}
 
 			desiredSub.SetGroupVersionKind(subscriptionGVK) // Create stripped this information
 
 			// Now it should match, so report Compliance
 			err = r.updateStatus(ctx, policy, createdCond("Subscription"), createdObj(desiredSub))
 			if err != nil {
-				return fmt.Errorf("error updating the status for a created Subscription: %w", err)
+				return desiredSub, fmt.Errorf("error updating the status for a created Subscription: %w", err)
 			}
 		}
 
-		return nil
+		return desiredSub, nil
```

Comment:
I think returning something like `nil, nil` would be better here if we can return the subscription in the `enforce` case. Then in my PR, I could use the return value to determine if the desired subscription exists on the cluster (`nil` vs a subscription object) and skip a call to `r.DynamicWatcher.Get` to retrieve the subscription.

Reply:
It's possible... but I think that if `inform` mode returned `nil, nil` here, it would artificially limit some of the information we can get. For example, it is generally possible to check if the CatalogSource specified in the policy exists on the cluster, even if the Subscription doesn't exist.

Reply:
I might be overlooking something but is the point of this to reduce calls to `buildSubscription`?

Reply:
Mostly. This would reduce calls to `buildSubscription` and `dynamicWatcher.Get` for the Subscription.

This can also return the "merged" subscription, regardless of inform/enforce, which hypothetically could be useful for some other healthcheck... I don't see a need for that yet, but it feels like it's easy enough to "do it right" now, so we don't have to worry about it later. For example if some other check used the `channel` in the subscription, this function would always return the *desired* setting based on the policy, which is what I think the user would be expecting.

Reply:
I see, I think that would be useful in general. For my specific case I'm only calling `dynamicWatcher.Get` on a catalogsource when the subscription doesn't exist yet, so I might need to end up calling `dynamicWatcher.Get` on the subscription anyways in `handleCatalogSource` because the return values of `handleSubscription` doesn't seem to indicate whether or not a subscription exists on the cluster.